From 8291f8144b5a47c13fdf9da4af1a9d9623a3ce7c Mon Sep 17 00:00:00 2001
From: Fushan Wen <qydwhotmail@gmail.com>
Date: Sun, 9 Feb 2025 00:20:44 +0000
Subject: [PATCH] core/outputlayer: guard against null m_output

SENTRY: KWIN-8ZG
FIXED-IN: 6.3.1


(cherry picked from commit c97bc26ca9de8b1462f6ccb05fb2dafe01cd82cb)

Co-authored-by: Fushan Wen <qydwhotmail@gmail.com>
---
 src/core/outputlayer.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/core/outputlayer.cpp b/src/core/outputlayer.cpp
index f041a8d1d9..9c4aaf2f50 100644
--- a/src/core/outputlayer.cpp
+++ b/src/core/outputlayer.cpp
@@ -50,7 +50,9 @@ QRegion OutputLayer::repaints() const
 void OutputLayer::addRepaint(const QRegion &region)
 {
     m_repaints += region;
-    m_output->renderLoop()->scheduleRepaint(nullptr, nullptr);
+    if (m_output) {
+        m_output->renderLoop()->scheduleRepaint(nullptr, nullptr);
+    }
 }
 
 void OutputLayer::resetRepaints()
-- 
2.48.1

